src/ImageRunModal.jsx

```diff
@@ -393,17 +419,43 @@ export class ImageRunModal extends React.Component {
 
                         for (const result of reply) {
                             if (result.status === "fulfilled") {
-                                imageResults = imageResults.concat(JSON.parse(result.value));
+                                try {
+                                    imageResults = imageResults.concat(JSON.parse(result.value));
+                                } catch (err) {
+                                    console.error(err);
+                                }
                             } else {
+                                // manifests query returns 404 Not Found in case used URL is not an image
+                                // supress this error as it will either result in "No images found" or prioritize other erros from search queries
+                                if (result.reason.status === 404) {
+                                    continue;
+                                }
+                                // Supress "could't search registry" error when local image is found
+                                if (result.reason.message.includes("couldn't search registry") && imageExistsLocally(value, this.props.localImages)) {
+                                    continue;
+                                }
                                 dialogError = _("Failed to search for new images");
                                 // TODO: add registry context, podman does not include it in the reply.
                                 dialogErrorDetail = result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images.");
                             }
                         }
+
                         // Group images on registry
                         const images = {};
                         imageResults.forEach(image => {
-                            // Add Tag is it's there
+                            // Manifests query
+                            if (image.manifests !== undefined) {
+                                // Only use manifests result if nothing else was found (non-searchable image repository)
+                                if (imageResults.length === 1) {
```

Comment:
This needs some further explanation in a comment. What is that single result which we are ignoring? Can this be asserted/checked?

Reply:
As mentioned in the comment above this is here to handle case where manifest search is successful but other searches fail and image isn't available locally.

Reply:
Another option that comes to mind is to filter out duplicate entries in `filterImages` rather than this seemingly confusing "ignoring manifest result" in specific scenarios

Reply:
I feel like it should be the *other* way around: If the user typed in something which is a valid manifest -- i.e. the string uniquely identifies a valid container -- then this should count as the one and only search result IMHO. Search should then be discarded (and most likely won't bring up anything else anyway?). However, that's only true if there is no tag: if I type "docker.io/ubuntu" it should show me tags like ":stable", ":devel" and so on.

So, tricky. 

But that code is still horribly confusing. First of all the length check happens *inside* of an iteration of that very list. And second, at that point the information is lost where it came from -- from the manifest or the search query. I think you should handle these two in completely separate code paths, they are just too different.